azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -509,13 +525,26 @@ public void cancelFlow(final int execId, final String user)
     this.flowRunner.kill(user);
   }
 
-  /**
-   * Attempts to retry the failed jobs in a running execution.
-   *
-   * @param execId
-   * @param user
-   * @throws ExecutorManagerException
-   */
+  @Override
+  public void cancelJobBySLA(final int execId, final String jobId)
+      throws ExecutorManagerException {
+    if (this.flowRunner == null) {
+      throw new ExecutorManagerException("Execution " + execId + " is not running.");
+    }
+
+    this.flowRunner.getExecutableFlow().setModifiedBy("SLA");
+
+    for (final JobRunner jobRunner : this.flowRunner.getActiveJobRunners()) {
+      if (jobRunner.getJobId().equals(jobId)) {
```

Comment:
Please put negative case here and put the positive case after that.
if (!jobRunner.getJobId().equals(jobId)) { 
  continue};
}
<rest of the code>

Reply:
Do you mean like this?
```
{ // start for loop
  if (!jobRunner.getJobId().equals(jobId)) {
    continue
  };
  logger.info("Killing job " + jobId + " in execution " + execId + " by SLA");
  jobRunner.killBySLA();
  break;
} // end for
```
I understand that asking for the negative case is better overall but given the simplicity of the logic (log and call kill action) existing code looks more readable and simple to me than above block. I can definitely make the change if you feel strongly about it though.

Reply:
This code may become larger if further development happens here for some reason in the future. It is better to keep it the way suggested for such cases.